scripts/CodeCoverage.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+
+import os
+import re
+import sys
+import argparse
+import shutil
+import zipfile
+import ntpath
+import subprocess
+
+DEFAULT_WHITELIST = ".*cpp$"
+DEFAULT_BLACKLIST = "/usr/local/probe/.*"
+
+CMAKEFILEPATH = "CMakeLists.txt"
+LIBRARY_TO_BUILD_REGEX = 'SET\(LIBRARY_TO_BUILD (.*)\)$'
+
+def get_file_content(filename):
+    with open(filename) as f:
+        content = f.readlines()
+        return content
+
+def get_library_to_build_name(line_str):
+    match_obj = re.search(LIBRARY_TO_BUILD_REGEX, line_str, re.M|re.I)
+    if match_obj:
+        return True, match_obj.group(1)
+    return False, ""
+
+def get_project_name_from_CMakeLists_file(full_file_path):
+    for line in get_file_content(filename=full_file_path):
+        found, project_name = get_library_to_build_name(line_str=line)
+        if found:
+            return True, project_name
+    return False, ""
+
+def print_error_and_usage(argParser, error):
+    print "Error:  " + error + "\n"
+    print argParser.print_help()
+    sys.exit(2)
+
+def santize_input_args(arg_parser, args):
+    if len(sys.argv) > 3:
+        print_error_and_usage(argParser=arg_parser, error="Too many arguments supplied.")
+
+def destroy_directory(dir_path):
+    shutil.rmtree(dir_path)
+
+def clean_and_build_directory(dir_path):
+    if os.path.exists(dir_path):
+        destroy_directory(dir_path=dir_path)
+    os.makedirs(dir_path)
+
+def remove_file_extension(filename):
+    return os.path.splitext(filename)[0]
+
+def strip_file_from_full_path(full_file_path):
+    return ntpath.basename(full_file_path)
+
+def unzip_file(full_file_path, directory_to_extract_to):
+    to_unzip = zipfile.ZipFile(full_file_path, 'r')
+    unzipped_dir_name = remove_file_extension(filename=strip_file_from_full_path(full_file_path=full_file_path))
+    full_unzipped_dir_path = directory_to_extract_to + "/" + unzipped_dir_name
+    if os.path.exists(full_unzipped_dir_path):
+        destroy_directory(full_unzipped_dir_path)
+    to_unzip.extractall(directory_to_extract_to)
+    to_unzip.close()
+
+
+def run_CMake_cmd():
+    CMAKE = '/usr/local/probe/bin/cmake'
```

Comment:
use consistent naming -- no need for CONSTANT style naming on strings in python.

``` 
cmake = 'foo'
```